superset/views/core.py

```diff
@@ -1562,17 +1564,22 @@ def copy_dash(self, dashboard_id):
                 session.add(new_slice)
                 session.flush()
                 new_slice.dashboards.append(dash)
-                old_to_new_sliceids[slc.id] = new_slice.id
+                old_to_new_sliceids['{}'.format(slc.id)] = \
+                    '{}'.format(new_slice.id)
 
             # update chartId of layout entities
-            for value in data['positions'].values():
-                if (
-                    isinstance(value, dict) and value.get('meta') and
-                    value.get('meta').get('chartId')
-                ):
-                    old_id = value.get('meta').get('chartId')
-                    new_id = old_to_new_sliceids[old_id]
-                    value['meta']['chartId'] = new_id
+            if is_v2_dash:
+                for value in data['positions'].values():
+                    if (
+                        isinstance(value, dict) and value.get('meta') and
+                        value.get('meta').get('chartId')
+                    ):
+                        old_id = '{}'.format(value.get('meta').get('chartId'))
```

Comment:
@graceguo-supercat can you add a comment about strings vs ints here so the logic is documented somewhere?

Reply:
in v2 dashboard position json, chartId is integer type:
```javascript
"DASHBOARD_CHART_TYPE-H10M97Xk-7": {
        "children": [],
        "id": "DASHBOARD_CHART_TYPE-H10M97Xk-7",
        "meta": {
            "chartId": 136,
            "height": 50,
            "sliceName": "Treemap",
            "width": 8
        },
        "type": "DASHBOARD_CHART_TYPE"
    },
```

in older dashboard position json data, slice_id is string type: 
```javascript
   {
        "slice_id": "49", 
        "size_x": 8, 
        "size_y": 16, 
        "v": 1, 
        "col": 33, 
        "row": 24
    }, 
```